openassessment/workflow/models.py

```diff
@@ -239,7 +240,8 @@ def get_score(self, assessment_requirements, step_for_name):
                     else:
                         requirements = assessment_requirements.get(assessment_step_name, {})
                     score = get_score_func(self.submission_uuid, requirements)
-                    break
+                    if score:
+                        break
```

Comment:
The score is only set when for each workflow step `assessment_completed_at` has been set, which is only set when the corresponding api returns `True` for `assessment_is_finished`, and for peer api that will only happen if the peers have assessed the submission. So as far as I understand it, it should work.

However, I think your suggestion will make the code clearer. So I'll update the `get_score` function to return a tuple [done, score.]